src/main/java/uk/gov/digital/ho/proving/income/api/NinoUtils.java

```diff
@@ -1,20 +1,48 @@
 package uk.gov.digital.ho.proving.income.api;
 
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
 import java.util.regex.Pattern;
 
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+
+@Component
 public class NinoUtils {
-    private NinoUtils() {
-        // utility
-    }
-    public static String sanitiseNino(String nino) {
-        return nino.replaceAll("\\s", "").toUpperCase();
+    private static final Pattern NINO_PATTERN = Pattern.compile("(^((?!(BG|GB|KN|NK|NT|TN|ZZ)|([DFIQUV])[A-Z]|[A-Z]([DFIOQUV]))[A-Z]{2})[0-9]{6}[A-D]?$)");
+
+    public String sanitise(final String nino) {
+        if (isNull(nino)) {
+            return null;
+        }
+        return StringUtils.deleteWhitespace(nino).toUpperCase();
     }
 
-    public static void validateNino(String nino) {
-        final Pattern pattern = Pattern.compile("^[a-zA-Z]{2}[0-9]{6}[a-dA-D]{1}$");
-        if (!pattern.matcher(nino).matches()) {
+    public void validate(final String nino) {
+        if (isInvalid(nino)) {
             throw new IllegalArgumentException("Error: Invalid NINO");
```

Comment:
While you're rewriting this class, can we swap this out for a dedicated runtime exception?

Reply:
I've done this - one additoinal thing I noticed was that we have an `IllegalArgumentException` handler that maps to `Bad Request`. Do we want this to be validation Illegal Argument Exceptions only?